php/admin/class-onboarding.php

```diff
@@ -0,0 +1,392 @@
+	 */
+	public function show_edit_block_notice() {
+		$example_post_id = get_option( 'block_lab_example_post_id' );
+
+		if ( ! $example_post_id ) {
+			return;
+		}
+		?>
+		<div class="block-lab-edit-block block-lab-notice notice">
+			<h2>👩‍🔬 <?php echo esc_html_e( 'Ready to begin?', 'block-lab' ); ?></h2>
+			<p class="intro">
+				<?php
+				echo wp_kses_post(
+					sprintf(
+						// translators: Placeholders are <strong> html tags.
+						__( 'We created this %1$sExample Block%2$s to show you just how easy it is to get started.', 'block-lab' ),
+						'<strong>',
+						'</strong>'
+					)
+				);
+				?>
+			</p>
+			<p>
+				<?php
+				echo wp_kses_post(
+					sprintf(
+						// translators: Placeholders are <strong> and <a> html tags.
+						__( 'You can %1$sEdit%2$s the block to learn more, or just %3$sTrash%4$s it to dismiss this message.', 'block-lab' ),
+						'<strong>',
+						'</strong>',
+						'<a href="' . get_delete_post_link( $example_post_id ) . '" class="trash">',
+						'</a>'
+					)
+				);
+				?>
+			</p>
+			<span class="pointer">☝️</span>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render the Add Fields message.
+	 */
+	public function show_add_fields_notice() {
+		$post  = get_post();
+		$block = new Block( $post->ID );
+
+		/**
+		 * We add 4 fields to our Example Block in add_dummy_data().
+		 */
+		if ( count( $block->fields ) > 4 ) {
+			return;
+		}
+		?>
+		<div class="block-lab-add-fields block-lab-notice">
+			<h2>🧐 <?php esc_html_e( 'Try adding a field.', 'block-lab' ); ?></h2>
+			<p><?php esc_html_e( 'Fields let you define the options you see when adding your block to a post.', 'block-lab' ); ?></p>
+			<span class="pointer">👈</span>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render the Add Fields message.
+	 */
+	public function show_publish_notice() {
+		$post  = get_post();
+		$block = new Block( $post->ID );
+
+		/**
+		 * We add 4 fields to our Example Block in add_dummy_data().
+		 */
+		if ( count( $block->fields ) > 4 ) {
+			return;
+		}
+		?>
+		<div class="block-lab-publish block-lab-notice">
+			<h2>🧪 <?php esc_html_e( 'Time to experiment!', 'block-lab' ); ?></h2>
+			<p class="intro"><?php esc_html_e( 'Choose an icon, change the category, and investigate a few different field types.', 'block-lab' ); ?></p>
+			<p>
+				<?php
+				echo wp_kses_post(
+					sprintf(
+						// translators: Placeholders are <strong> html tags.
+						__( 'When you\'re ready, save your block by pressing %1$sPublish%2$s.', 'block-lab' ),
+						'<strong>',
+						'</strong>'
+					)
+				);
+				?>
+			</p>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render the Add to Post message.
+	 */
+	public function show_add_to_post_notice() {
+		$post = get_post();
+
+		if ( ! $post || ! isset( $post->post_name ) || empty( $post->post_name ) ) {
+			return;
+		}
+
+		if ( 'publish' !== $post->post_status ) {
+			return;
+		}
+
+		$template = block_lab_locate_template( 'blocks/block-' . $post->post_name . '.php', '', true );
```

Comment:
This is fine as is, but I might have used:

```php
"blocks/block-"{$post->post_name}.php"
```

...instead of:

```php
'blocks/block-' . $post->post_name . '.php'
```

It's just a personal preference, though.

Reply:
Fixed in 2705c6c.